lib/schemaUtils.js

```diff
@@ -2779,19 +2779,131 @@ module.exports = {
+    }
+
+    _.forEach(headers, (header, index) => {
+      if (_.toLower(header.key) === 'content-type') {
+        contentHeader = header;
+        contentHeaderIndex = index;
+        return false;
+      }
+    });
+
+    // Schema body content has no media type objects
+    if (!_.isEmpty(contentHeader) && _.isEmpty(mediaTypes)) {
+      // ignore mismatch for default header (text/plain) added by conversion
+      if (options.showMissingInSchemaErrors && _.toLower(contentHeader.value) !== TEXT_PLAIN) {
+        mismatches.push({
+          property: mismatchProperty,
+          transactionJsonPath: transactionPathPrefix + `[${contentHeaderIndex}]`,
+          schemaJsonPath: null,
+          reasonCode: 'MISSING_IN_SCHEMA',
+          // Reason for missing in schema suggests that certain media type in req/res body is not present
+          reason: `The ${mismatchProperty === 'HEADER' ? 'request' : 'response'} body should have content of` +
+            ` Media type "${contentHeader.value}"`
+        });
+      }
+    }
+
+    // No request/response content-type header
+    else if (_.isEmpty(contentHeader) && !_.isEmpty(mediaTypes)) {
+      let mismatchObj = {
+        property: mismatchProperty,
+        transactionJsonPath: transactionPathPrefix,
+        schemaJsonPath: schemaPathPrefix,
+        reasonCode: 'MISSING_IN_REQUEST',
+        reason: `The ${humanPropName} "Content-Type" was not found in the transaction`
+      };
+
+      if (options.suggestAvailableFixes) {
+        mismatchObj.suggestedFix = {
+          key: 'Content-Type',
+          actualValue: null,
+          suggestedValue: {
+            key: 'Content-Type',
+            value: suggestedContentHeader
+          }
+        };
+      }
+      mismatches.push(mismatchObj);
+    }
+
+    // Invalid type of header found
+    else if (!_.isEmpty(contentHeader) && !_.includes(mediaTypes, contentHeader.value)) {
+      let mismatchObj = {
+        property: mismatchProperty,
+        transactionJsonPath: transactionPathPrefix + `[${contentHeaderIndex}].value`,
+        schemaJsonPath: schemaPathPrefix,
+        reasonCode: 'INVALID_TYPE',
+        reason: `The ${humanPropName} "Content-Type" needs to be "${suggestedContentHeader}",` +
+          ` but we found "${contentHeader.value}" instead`
+      };
+
+      if (options.suggestAvailableFixes) {
+        mismatchObj.suggestedFix = {
+          key: 'Content-Type',
+          actualValue: contentHeader.value,
+          suggestedValue: suggestedContentHeader
+        };
+      }
+      mismatches.push(mismatchObj);
+    }
+    return mismatches;
+  },
+
+  checkRequestHeaders: function (headers, transactionPathPrefix, schemaPathPrefix, schemaPath,
+    components, options, schemaCache, callback) {
     let schemaHeaders = _.filter(schemaPath.parameters, (param) => { return param.in === 'header'; }),
+      reqHeaders = _.filter(headers, (header) => { return _.toLower(_.get(header, 'key')) !== 'content-type'; }),
```

Comment:
This needs to have a comment explaining why you're excluding content-type
Ideally, the list of 'implicit headers' should be an array specified elsewhere. There will be more you add to this list later.